examples/pytorch/graphsage/experimental/train_dist_transductive.py

```diff
@@ -0,0 +1,355 @@
+import os
+os.environ['DGLBACKEND']='pytorch'
+from multiprocessing import Process
+import argparse, time, math
+import numpy as np
+from functools import wraps
+import tqdm
+
+import dgl
+from dgl import DGLGraph
+from dgl.data import register_data_args, load_data
+from dgl.data.utils import load_graphs
+import dgl.function as fn
+import dgl.nn.pytorch as dglnn
+from dgl.distributed import DistDataLoader
+from dgl.distributed.nn import NodeEmbedding
+from dgl.distributed.optim import SparseAdagrad
+
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+import torch.multiprocessing as mp
+from torch.utils.data import DataLoader
+
+from train_dist import DistSAGE, compute_acc
+
+def load_subtensor(g, seeds, input_nodes, device):
+    """
+    Copys features and labels of a set of nodes onto GPU.
+    """
+    batch_labels = g.ndata['labels'][seeds].to(device)
+    return batch_labels
+
+class NeighborSampler(object):
+    def __init__(self, g, fanouts, sample_neighbors, device):
+        self.g = g
+        self.fanouts = fanouts
+        self.sample_neighbors = sample_neighbors
+        self.device = device
+
+    def sample_blocks(self, seeds):
+        seeds = th.LongTensor(np.asarray(seeds))
+        blocks = []
+        for fanout in self.fanouts:
+            # For each seed node, sample ``fanout`` neighbors.
+            frontier = self.sample_neighbors(self.g, seeds, fanout, replace=True)
+            # Then we compact the frontier into a bipartite graph for message passing.
+            block = dgl.to_block(frontier, seeds)
+            # Obtain the seed nodes for next layer.
+            seeds = block.srcdata[dgl.NID]
+
+            blocks.insert(0, block)
+
+        input_nodes = blocks[0].srcdata[dgl.NID]
+        seeds = blocks[-1].dstdata[dgl.NID]
+        batch_labels = load_subtensor(self.g, seeds, input_nodes, "cpu")
+        blocks[-1].dstdata['labels'] = batch_labels
+        return blocks
+
+class TransDistSAGE(DistSAGE):
+    def __init__(self, in_feats, n_hidden, n_classes, n_layers,
+                 activation, dropout):
+        super(TransDistSAGE, self).__init__(in_feats, n_hidden, n_classes, n_layers, activation, dropout)
+
+    def inference(self, standalone, g, x, batch_size, device):
```

Comment:
is this function different for inductive and transductive models?

Reply:
transductive do not need to copy node features.